core/templates/components/state-editor/state-editor-properties-services/state-editor.service.spec.ts

```diff
@@ -145,5 +232,342 @@ describe('Editor state service', () => {
+    let newSolution = sof.createFromBackendDict({
+      answer_is_exclusive: true,
+      correct_answer: 'test_answer_new',
+      explanation: {
+        content_id: '2',
+        html: 'test_explanation1_new',
+      },
+    });
+    ecs.setInteraction(mockInteraction);
+    expect(ecs.interaction.solution).toEqual(sof.createFromBackendDict({
+      answer_is_exclusive: true,
+      correct_answer: 'test_answer',
+      explanation: {
+        content_id: '2',
+        html: 'test_explanation1',
+      },
+    }));
+    ecs.setInteractionSolution(newSolution);
+    expect(ecs.interaction.solution).toEqual(newSolution);
+  });
+
+  it('should set interaction hints', () => {
+    let newHints = [hof.createFromBackendDict({
+      hint_content: {
+        content_id: '',
+        html: 'This is a hint'
+      }
+    })];
+    ecs.setInteraction(mockInteraction);
+    expect(ecs.interaction.hints).toEqual([]);
+    ecs.setInteractionHints(newHints);
+    expect(ecs.interaction.hints).toEqual(newHints);
+  });
+
+  it('should set in question mode', () => {
+    expect(ecs.isInQuestionMode()).toBe(null);
+    ecs.setInQuestionMode(true);
+    expect(ecs.isInQuestionMode()).toBe(true);
+    ecs.setInQuestionMode(false);
+    expect(ecs.isInQuestionMode()).toBe(false);
+  });
+
+  it('should set correctness feedback enabled', () => {
+    expect(ecs.getCorrectnessFeedbackEnabled()).toBe(null);
+    ecs.setCorrectnessFeedbackEnabled(true);
+    expect(ecs.getCorrectnessFeedbackEnabled()).toBe(true);
+    ecs.setCorrectnessFeedbackEnabled(false);
+    expect(ecs.getCorrectnessFeedbackEnabled()).toBe(false);
+  });
+
+  it('should set inapplicable skill misconception ids', () => {
+    expect(ecs.getInapplicableSkillMisconceptionIds()).toEqual([]);
+    ecs.setInapplicableSkillMisconceptionIds(['id1', 'id2']);
+    expect(ecs.getInapplicableSkillMisconceptionIds()).toEqual(['id1', 'id2']);
+  });
+
+  it('should check if current solution is valid', () => {
+    // Set 'Hola' as the active state.
+    ecs.activeStateName = 'Hola';
+    // At present, we are not keeping track of the solution's validity. So, we
+    // initialize the Solution Validity Service with the state. Upon,
+    // initialization the solution validity is set as true.
+    expect(ecs.isCurrentSolutionValid()).toBe(undefined);
+    solutionValidityService.init(['Hola']);
+    expect(ecs.isCurrentSolutionValid()).toBe(true);
+  });
+
+  it('should delete current solution validity', () => {
+    ecs.activeStateName = 'Hola';
+    expect(ecs.isCurrentSolutionValid()).toBe(undefined);
+    solutionValidityService.init(['Hola']);
+    expect(ecs.isCurrentSolutionValid()).toBe(true);
+    ecs.deleteCurrentSolutionValidity();
+    expect(ecs.isCurrentSolutionValid()).toBe(undefined);
+  });
+
+  it('should get event emitter on state editor initialization', () => {
+    spyOn(ecs.onStateEditorInitialized, 'subscribe');
+    ecs.onStateEditorInitialized.subscribe();
```

Comment:
Here and below: It looks like we are spying on X, calling X directly in the test and then expecting it to have been called. Is this expected?

Hm, actually, I don't think we need to test these getters TBH. In some service spec files we seem to have simply subscribed to such getter and then unsubscribed to them in the afterAll block (see `router.service.spec.ts`, `questions-list.service.spec.ts`). Should we do a similar thing?

Reply:
Done